src/Symfony/Component/Messenger/EventListener/SendFailedMessageForRetryListener.php

```diff
@@ -21,13 +21,16 @@
 use Symfony\Component\Messenger\Retry\RetryStrategyInterface;
 use Symfony\Component\Messenger\Stamp\DelayStamp;
 use Symfony\Component\Messenger\Stamp\RedeliveryStamp;
+use Symfony\Component\Messenger\Stamp\StampInterface;
 use Symfony\Component\Messenger\Transport\Sender\SenderInterface;
 
 /**
  * @author Tobias Schultze <http://tobion.de>
  */
 class SendFailedMessageForRetryListener implements EventSubscriberInterface
 {
+    private const HISTORY_SIZE = 10;
```

Comment:
I think it'd be better if this could be configurable via a constructor parameter (which can have a default value of 10).

Reply:
I see you've added the constructor parameter but it is not configurable at the moment. I think you should add the option to the Messenger config tree and then replace the constructor argument in the framework bundle extension.

Reply:
I don't think it worth a configuration option.
If people expose a real world use case where they need more/less history, we could open a dedicated PR